### PR TITLE
Refactor/documentation and testcases

### PR DESCRIPTION
**Pull request by bstien**
# What
- Added docs for `diff(old:new:comparator`
- Added some new testcases (sanity checks)
- Renamed testcases
- Moved `IndexPathResult` to folder `Models`

### CODE REVIEW

File: DiffuseTests/DiffuseComparatorTests.swift [hunk 1]
```diff
@@ -5,13 +5,42 @@
...
+        let new = [Int]()
+        let changes = Diffuse.diff(old: old, new: new, comparator: { $0 == $1 })
+
+        // Both collections are empty.
+        XCTAssertEqual(0, changes.count)
+    }
+
+    // MARK: - Comparing primitives
+
+    func testEmptyOld_withPrimitives() {
```
Comment (3lvis) [inline DiffuseTests/DiffuseComparatorTests.swift]: I would try to avoid using underscores for naming in iOS

Comment (bstien) [reply DiffuseTests/DiffuseComparatorTests.swift]: I agree. The names of these testing functions aren't long enough (yet) to justify the underscore. Changed

**Review by 3lvis — APPROVED** [review]
💥💥💥

File: DiffuseTests/DiffuseHashableTests.swift [hunk 2]
```diff
@@ -120,11 +168,23 @@ class DiffuseHashableTests: XCTestCase {
...
+        //      = 2 changes (updates)
+        //        Index 0 and 1 are both considered updated.
+        //
+        //        "But, why? `A` is removed, `B` is updated and moved to index 0 and `D` is inserted at index 1, this should be
+        //        one delete, one update, one move and one insert, right?"
+        //
+        //        Nope.
+        //        We're using `hashValue` to compare elements, so `B` is actually considered to be a new element since it's
+        //        `hashValue` has changed because of the update. This means that the algorithm now thinks two deletions and
+        //        two insertions has happened instead. Both `A` and `old B` are considered removed from index 0 and 1, and new
```
Comment (vadymmarkov) [inline DiffuseTests/DiffuseHashableTests.swift]: Should it be:
```suggestion
        //        two insertions have happened instead. Both `A` and `old B` are considered removed from index 0 and 1, and new
```

**Review by vadymmarkov — APPROVED** [review]
👏 👏 👏

**Pull request merged**